### PR TITLE
RavenDB-16460 Return free disk space back to the operating system

### PR DESCRIPTION
**Pull request by ayende**
### Issue link

https://issues.hibernatingrhinos.com/issue/[RavenDB-16460

### Additional description

We'll now keep track and free (using sparse files) disk space back to the operating system.

### Type of change

- [x] New feature

### How risky is the change?

- [x] Moderate 

### Backward compatibility

- [x] Non breaking change

### Is it platform specific issue?

- [x] No

### Documentation update

- [x] Need to document the behavior

### Testing by Contributor

- [x] Tests have been added that prove the fix is effective or that the feature works

### Testing by RavenDB QA team

- [x] This change requires a special QA testing due to possible performance or resources usage implications (CPU, memory, IO). Please mark the issue on YouTrack using `QA Required` tag.

### Is there any existing behavior change of other features due to this change?

- [x] No

### UI work

- [x] It requires further work in the Studio. Please mark the issue on YouTrack using `Studio Required` tag.

### CODE REVIEW

File: src/Raven.Pal/inc/rvn.h [hunk 1]
```diff
@@ -37,6 +37,13 @@ enum
     OPEN_FILE_DO_NOT_MAP = (1<<9)
 };
 
+enum
```
Comment (ppekrol) [inline src/Raven.Pal/inc/rvn.h:40]: @ayende NuGet is aching packages, you need to update the version of the Raven.PAL, it now goes from cache on CI

Comment (ayende) [reply src/Raven.Pal/inc/rvn.h:40]: Not a problem for me, just means that on release, we'll not have 6.2.0 but something like 6.2.138

File: test/FastTests/Voron/Trees/FreeSpaceTest.cs [hunk 2]
```diff
@@ -246,5 +247,81 @@ public void CanGetListOfAllFreedPages(int maxPageNumber, int numberOfFreedPages,
                 Assert.Equal(sorted, retrievedFreePages);
             }
         }
+
+
+        [Fact]
```
Comment (ppekrol) [inline test/FastTests/Voron/Trees/FreeSpaceTest.cs]: This should fail AllTestsShouldUseRavenFactOrRavenTheoryAttributes test?

**Review by ppekrol — APPROVED** [review]
By glance LGTM, but we need to check Linux failures

File: test/FastTests/Voron/Trees/FreeSpaceTest.cs (hunk 2 above)
Comment (ayende) [reply test/FastTests/Voron/Trees/FreeSpaceTest.cs]: Fixed

**Pull request merged**